app/controllers/posts_controller.rb

```diff
@@ -707,6 +713,16 @@ def find_post_from_params_by_number
     find_post_using(by_number_finder)
   end
 
+  def find_post_from_params_by_date
+    by_date_finder = Post
```

Comment:
You'll need to initialize an instance of a `TopicView` here and pass it the current filters for the topic. Otherwise, this query may return a post that is not within the stream.

Reply:
So what we do in:

find_post_from_params_by_number
find_post_from_params

is also wrong?